eventuals/task.h

```diff
@@ -17,22 +18,15 @@ namespace detail {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename E_, typename Value_>
+template <typename E_, typename From_, typename To_>
 struct HeapTask {
-  // NOTE: we take 'Value_' as a template argument rather than
-  // computing it from 'E_' because that's the type that we should
-  // expect for our callbacks. We make sure when we construct a 'Task'
-  // that 'E_::ValueFrom<void>' is convertible to 'Value_' so we know
-  // that what ever 'E_' actually produces we can pass it along to the
-  // callbacks.
-
   template <typename Arg_>
```

Comment:
Seems like `Arg_` was never actually used before! So let's just remove it and use the enclosing `To_`!
```suggestion
```